seed/SeedOsc/SeedOsc.cpp

```diff
@@ -0,0 +1,101 @@
+#include "daisy_seed.h"
+#include "daisysp.h"
+
+// Use the daisy namespace to prevent having to type 
+// daisy:: before all libdaisy functions
+using namespace daisy;
+using namespace daisysp;
+
+// Declare a DaisySeed object called hardware
+DaisySeed hardware;
+Oscillator osc;
+AdEnv env;
+
+Switch button1;
+
+void AudioCallback(float* in, float* out, size_t size)
+{
+    float osc_out, env_out;
+
+    //Nobody likes a bouncy button
+    button1.Debounce();
+
+    //If you push the button,...
+    if (button1.RisingEdge())
+    {
+      env.Trigger(); //Trigger the envelope!
+    }
+
+    //Convert floating point knob to midi (0-127)
+    //Then convert midi to freq. in Hz
+    osc.SetFreq(mtof(hardware.adc.GetFloat(0) * 127));
+
+    //Fill the block with samples
+    for (size_t i = 0; i < size; i+=2)
+    {
+        //Get the next envelope value
+        env_out = env.Process();
+	//Set the oscillator volume to the latest env value
+	osc.SetAmp(env_out);
+	//get the next oscillator sample
+	osc_out = osc.Process();
+
+	//Set the left and right outputs
+	out[i] = osc_out;
+	out[i+1] = osc_out;
+    }
+}
+
+
+int main(void)
+{
+    // Configure and Initialize the Daisy Seed
+    // These are separate to allow reconfiguration of any of the internal
+    // components before initialization.
+    hardware.Configure();
+    hardware.Init();
+    hardware.SetAudioBlockSize(48);
+
+    //How many samples in one block of audio. We fill one block every callback
+    float blocksize = 48;
+    
+    //How many samples we'll output per second
+    float samplerate = hardware.AudioSampleRate();
+
+    // (sample / second) / (samples / callback) = callbacks / second
+    // Each callback fills one block, so here block = callback.
+    float callbackrate = samplerate / blocksize;
+    
+    //Create and ADC configuration
+    AdcChannelConfig adcConfig;
+    //Add pin 21 as an analog input in this config. We'll use this to read the knob
+    adcConfig.InitSingle(hardware.GetPin(21));
+
+    //Set the ADC to use our configuration
+    hardware.adc.Init(&adcConfig, 1);
+    //Start the adc
+    hardware.adc.Start();
```

Comment:
It's probably best to start all the things at about the same time. So let's move this down to right before `StartAudio`